eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -97,15 +97,11 @@ object Channel {
    */
   case class OutgoingMessage(msg: LightningMessage, peerConnection: ActorRef)
 
-  def failTimedout(adds: Set[UpdateAddHtlc], relayer: ActorRef, commits: AbstractCommitments, log: akka.event.LoggingAdapter): Unit =
+  def failHtlcs(adds: Set[UpdateAddHtlc], commits: AbstractCommitments, log: akka.event.LoggingAdapter)(onFound: (Origin, UpdateAddHtlc) => Unit): Unit =
```

Comment:
This goes a bit further since failing timed out and overridden HTLCs code is similar. If this is too much I'll undo this change.